src/Module/Contact/Conversations.php

```diff
@@ -0,0 +1,117 @@
+<?php
+/**
+ * @copyright Copyright (C) 2010-2021, the Friendica project
+ *
+ * @license   GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace Friendica\Module\Contact;
+
+use Friendica\App;
+use Friendica\BaseModule;
+use Friendica\Contact\LocalRelationship\Repository\LocalRelationship;
+use Friendica\Content\Conversation;
+use Friendica\Content\Nav;
+use Friendica\Content\Widget;
+use Friendica\Core\L10n;
+use Friendica\Core\Protocol;
+use Friendica\Core\Theme;
+use Friendica\Database\DBA;
+use Friendica\Model;
+use Friendica\Module\Contact;
+use Friendica\Module\Security\Login;
+use Friendica\Network\HTTPException\NotFoundException;
+
+/**
+ *  Manages and show Contacts and their content
+ */
+class Conversations extends BaseModule
+{
+	/**
+	 * @var App\Page
+	 */
+	private $page;
+	/**
+	 * @var Conversation
+	 */
+	private $conversation;
+	/**
+	 * @var App\BaseURL
+	 */
+	private $baseUrl;
+	/**
+	 * @var LocalRelationship
+	 */
+	private $localRelationship;
+
+	public function __construct(L10n $l10n, LocalRelationship $localRelationship, App\BaseURL $baseUrl, App\Page $page, Conversation $conversation, array $parameters = [])
+	{
+		parent::__construct($l10n, $parameters);
+
+		$this->page = $page;
+		$this->conversation = $conversation;
+		$this->baseUrl = $baseUrl;
+		$this->localRelationship = $localRelationship;
+	}
+
+	public function content(): string
+	{
+		if (!local_user()) {
+			return Login::form($_SERVER['REQUEST_URI']);
+		}
+
+		// Backward compatibility: Ensure to use the public contact when the user contact is provided
+		// Remove by version 2022.03
+		$data = Model\Contact::getPublicAndUserContactID(intval($this->parameters['id']), local_user());
+		if (empty($data)) {
+			throw new NotFoundException($this->t('Contact not found.'));
+		}
+
+		$contact = Model\Contact::getById($data['public']);
+		if (!DBA::isResult($contact)) {
+			throw new NotFoundException($this->t('Contact not found.'));
+		}
+
+		// Don't display contacts that are about to be deleted
+		if (DBA::isResult($contact) && (!empty($contact['deleted']) || !empty($contact['network']) && $contact['network'] == Protocol::PHANTOM)) {
```

Comment:
```suggestion
		if ($this->dba->isResult($contact) && (!empty($contact['deleted']) || !empty($contact['network']) && $contact['network'] == Protocol::PHANTOM)) {
```
 And add it as dependency :)

Reply:
Nah, not just for `isResult` which has nothing to do with the database. In this condition, it should even be removed given the fact we are checking the existence of the array keys with `empty()`.

Reply:
Removed.